homework/list.js

```diff
@@ -6,17 +6,24 @@
   var _wrapper = document.querySelector('#note-list-wrapper');
 
   function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+    var li = new list();
+    this.fetchList()
+	.then((function(data) {
```

Comment:
You should practice to compose a Promise chain by multiple `then` and `catch`. And you should note that `catch` should **log AND re-throw** the error, or the following chains may hide the message, and then continuously perform some actions that should be ended after one exception occurs.